editor/sidebar/featured-image/index.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { __ } from 'i18n';
+import Button from 'components/button';
+import PanelBody from 'components/panel/body';
+import MediaUploadButton from 'blocks/media-upload-button';
+import Spinner from 'components/spinner';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+class FeaturedImage extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			media: null,
+			loading: false,
+		};
+	}
+
+	componentDidMount() {
+		this.fetchMedia();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.featuredImageId !== this.props.featuredImageId ) {
+			this.fetchMedia();
+		}
+	}
+
+	componentWillUnmount() {
+		if ( this.fetchMediaRequest ) {
+			this.fetchMediaRequest.abort();
+		}
+	}
+
+	fetchMedia() {
+		this.setState( { media: null } );
+		if ( ! this.props.featuredImageId ) {
+			this.setState( { loading: false } );
+			return;
+		}
+		this.setState( { loading: true } );
+		const mediaIdToLoad = this.props.featuredImageId;
+		this.fetchMediaRequest = new wp.api.models.Media( { id: mediaIdToLoad } ).fetch()
+			.done( ( media ) => {
+				if ( this.props.featuredImageId !== mediaIdToLoad ) {
+					return;
+				}
+				this.setState( {
+					loading: false,
+					media,
+				} );
+			} )
+			.fail( () => {
+				if ( this.props.featuredImageId !== mediaIdToLoad ) {
+					return;
+				}
+				this.setState( {
+					loading: false,
+				} );
+			} );
+	}
+
+	render() {
+		const { featuredImageId, onUpdateImage, onRemoveImage } = this.props;
+		const { media, loading } = this.state;
+
+		return (
+			<PanelBody title={ __( 'Featured image' ) } initialOpen={ false }>
+				<div className="editor-featured-image__content">
+					{ !! featuredImageId &&
+						<MediaUploadButton
+							buttonProps={ { className: 'button-link' } }
+							onSelect={ onUpdateImage }
+							type="image"
+						>
+							{ media &&
+								<img
```

Comment:
One minor thing we might consider is using the media's dimensions to assign a height to the space that the image will occupy, to avoid the jarring appearance of content being pushed down only after the image has loaded.

Height is difficult to calculate unless we know the width of the container, which we probably don't want to assume. We could use a technique for proportionally scaled responsive containers though, using padding to dynamically fill height.

See: 

- Demo: https://codepen.io/aduth/pen/GmLMqe
- Intrinsic Ratios: https://alistapart.com/article/creating-intrinsic-ratios-for-video